pkg/api/bzz.go

```diff
@@ -249,8 +249,16 @@ func (s *server) bzzDownloadHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	ctx := r.Context()
 
-	nameOrHex := mux.Vars(r)["address"]
-	pathVar := mux.Vars(r)["path"]
+	nameOrHex := chi.URLParam(r, "address")
+
+	url := strings.Split(strings.TrimSpace(r.URL.Path), "/")
+	if url[1] == "bzz" {
+		url = url[3:]
+	} else {
+		url = url[4:]
+	}
+	pathVar := strings.Join(url, "/")
+
```

Comment:
Generally I think `chi` offers a cleaner approach to routing, but I'm not sure what the new logic above is doing?

Reply:
aha! so the regex url variable only works for one subroute, meaning if you have a url like this:
`/bzz/aaa/lorem/ipsum.txt` and your path is like this `/bzz/{address}/{path:.*}`, the chi regex variable path only captures the lorem part, this worked right out of the box for gorilla/mux, but not chi.